lib/cfndsl/specification.rb

```diff
@@ -127,13 +132,20 @@ def self.extract_from_resource_spec!
           to_patch = JSON.parse(File.read(patch))
           to_patch.each_key do |type|
             to_patch[type].each_key do |primitive|
-              if primitive == 'patch'
-                jpatch = Hana::Patch.new to_patch[type]['patch']['operations']
-                jpatch.apply(spec_file[type])
-              else
-                jpatch = Hana::Patch.new to_patch[type][primitive]['patch']['operations']
-                jpatch.apply(spec_file[type][primitive])
+              # rubocop:disable Lint/HandleExceptions
+              begin
+                if primitive == 'patch'
+                  jpatch = Hana::Patch.new to_patch[type]['patch']['operations']
+                  jpatch.apply(spec_file[type])
+                else
+                  jpatch = Hana::Patch.new to_patch[type][primitive]['patch']['operations']
+                  jpatch.apply(spec_file[type][primitive])
+                end
+              rescue Hana::Patch::MissingTargetException
```

Comment:
Some of the patches are no longer required but the embedded resource spec does need them.  I think this is the right thing to do even if updating the spec, because you can assume that patches are going to be fixed upstream at some point.

Reply:
Oh - except I did delete the VPCEndpoint patch.  Probably should restore that.

Reply:
> because you can assume that patches are going to be fixed upstream at some point

We hope so... See CDK for patches that are still being maintained: https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/cfnspec/spec-source

> Oh - except I did delete the VPCEndpoint patch. Probably should restore that.

Please do restore

Reply:
Or... your way of handling rules supersedes this?

Reply:
The other option is to update the base spec, then remove unnessary patches. Probably needs a rake task and some optionally of whether to ignore patch errors.

Reply:
That sounds scary...

Reply:
OK, we already have a test that calls this method directly so we can just pass a parameter to allow the exception to be raised during the test (ie patches must all be valid for the embedded spec), but otherwise ignored if there are errors patching a user provided spec.
To test this behaviour I tried to use the latest spec but AWS have gone and broken it (property types with Type = List and no Properties - resulting in nonsense List of List of Type things - see AWSLakeFormationDataLakeSettingsAdmins)  I'm going to give this a couple of days to see if any movement on AWS side before implementing workarounds.